prometheus/testutil/testutil.go

```diff
@@ -153,6 +152,33 @@ func GatherAndCount(g prometheus.Gatherer, metricNames ...string) (int, error) {
 	return result, nil
 }
 
+// ScrapeAndCompare calls a remote exporter's endpoint which is expected to return some metrics in
+// plain text format. Then it compares it with the results that the `expected` would return.
+// If the `metricNames` is not empty it would filter the comparison only to the given metric names.
+func ScrapeAndCompare(url string, expected io.Reader, metricNames ...string) error {
+	resp, err := http.Get(url)
+	if err != nil {
+		return fmt.Errorf("scraping metrics failed: %s", err)
+	}
+
+	if resp.StatusCode != http.StatusOK {
```

Comment:
Shoudn't we ensure that we close the body here?
```suggestion
	defer resp.Body.Close()
	if resp.StatusCode != http.StatusOK {
```